### PR TITLE
WebUI: Making VOMS Message configurable Fix #1109

### LINKED ISSUE #1109: Error message for unregistered users in webui is ATLAS specific
Motivation
----------
The error message shown to unregistered users in the webui always points to ATLAS voms.

Modification
------------
Make the message configurable.

### PR DESCRIPTION
**Pull request by DhruvSondhi**
Making VOMS message configurable according to the needs 
------------------

This PR aims to fix & implement the configurable nature of the VOMS message thus obtained Fix #1109

### CODE REVIEW

File: lib/rucio/web/ui/flask/common/utils.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def x509token_auth(data=None):
         msg = "<br><br>Your certificate (%s) is not mapped to (possibly any) rucio account: %s at VO: %s." % (html_escape(dn), html_escape(ui_account), html_escape(ui_vo))
     else:
         msg = "<br><br>Your certificate (%s) is not mapped to (possibly any) rucio account: %s." % (html_escape(dn), html_escape(ui_account))
-    msg += "<br><br><font color=\"red\">First, please make sure it is correctly registered in <a href=\"https://voms2.cern.ch:8443/voms/atlas\">VOMS</a> and be patient until it has been fully propagated through the system.</font>"
+    msg += "<br><br><font color=\"red\">First, please make sure it is correctly registered in <a href=\"https://voms2.cern.ch:8443/voms/atlas\">| VOMS |</a> and be patient until it has been fully propagated through the system.</font>"
```
Comment (DhruvSondhi) [inline lib/rucio/web/ui/flask/common/utils.py]: Need some more information regarding the configurable nature of the message 🤔

Comment (bari12) [reply lib/rucio/web/ui/flask/common/utils.py]: I have added a quick comment to the issue about this #1109

File: lib/rucio/web/ui/flask/common/utils.py [hunk 2]
```diff
@@ -360,8 +360,11 @@ def x509token_auth(data=None):
         msg = "<br><br>Your certificate (%s) is not mapped to (possibly any) rucio account: %s at VO: %s." % (html_escape(dn), html_escape(ui_account), html_escape(ui_vo))
     else:
         msg = "<br><br>Your certificate (%s) is not mapped to (possibly any) rucio account: %s." % (html_escape(dn), html_escape(ui_account))
-    msg += "<br><br><font color=\"red\">First, please make sure it is correctly registered in <a href=\"https://voms2.cern.ch:8443/voms/atlas\">VOMS</a> and be patient until it has been fully propagated through the system.</font>"
-    msg += "<br><br>Then, if it is still not working please contact <a href=\"mailto:atlas-adc-ddm-support@cern.ch\">DDM Support</a>."
+        if ui_account == "ATLAS":
```
Comment (DhruvSondhi) [inline lib/rucio/web/ui/flask/common/utils.py]: Need to check for the actual account value being passed in the `ui_account` variable for `ATLAS` specificity

File: lib/rucio/web/ui/flask/common/utils.py [hunk 3]
```diff
@@ -360,8 +360,11 @@ def x509token_auth(data=None):
...
+            msg += "<br><br><font color=\"red\">First, please make sure it is correctly registered in <a href=\"https://voms2.cern.ch:8443/voms/atlas\">VOMS</a> and be patient until it has been fully propagated through the system.</font>"
+            msg += "<br><br>Then, if it is still not working please contact <a href=\"mailto:atlas-adc-ddm-support@cern.ch\">DDM Support</a>."
+        else:
```
Comment (DhruvSondhi) [inline lib/rucio/web/ui/flask/common/utils.py:370]: This needs to be reviewed for tailoring it to a particular part of the organisation

**Review by DhruvSondhi — COMMENTED** [review]
Reviewing current Status

**Comment by DhruvSondhi** [comment]
@bari12 If I can be provided with all the emails for the respective organisations, I can make a dictionary & hence we can tailor it according to the account type that is being passed 🤔

**Comment by bari12** [comment]
That won't really work, since we don't know about these yet and also don't want to manage these organisation details upstream. They should just be able to configure this in their deployment and tailor it to their needs.

**Comment by bari12** [comment]
This is still not correct, it should be read from a config file and not have hardcoded conditions in the file. Maybe @tbeerman can give you a hint.

File: lib/rucio/web/ui/flask/common/utils.py [hunk 4]
```diff
@@ -360,8 +360,12 @@ def x509token_auth(data=None):
         msg = "<br><br>Your certificate (%s) is not mapped to (possibly any) rucio account: %s at VO: %s." % (html_escape(dn), html_escape(ui_account), html_escape(ui_vo))
     else:
         msg = "<br><br>Your certificate (%s) is not mapped to (possibly any) rucio account: %s." % (html_escape(dn), html_escape(ui_account))
-    msg += "<br><br><font color=\"red\">First, please make sure it is correctly registered in <a href=\"https://voms2.cern.ch:8443/voms/atlas\">VOMS</a> and be patient until it has been fully propagated through the system.</font>"
-    msg += "<br><br>Then, if it is still not working please contact <a href=\"mailto:atlas-adc-ddm-support@cern.ch\">DDM Support</a>."
+        if "None" in config_get("client", "ca_cert"):
```
Comment (DhruvSondhi) [inline lib/rucio/web/ui/flask/common/utils.py]: I did this on the basis that only the client was being passed a `CA_Certificate` location ...I can do it on any other basis though but this is was evident :)

**Comment by tbeerman** [comment]
Please check https://github.com/rucio/rucio/commit/9a8cc9a17ba7a88c6b2fffe49a10ef4cb265f432

I've already implemented the configurable error message for webpy there. You just need to copy/paste for flask.

**Comment by DhruvSondhi** [comment]
Thank You Very Much @tbeerman for the input ... I was able to figure some part of the problem but not all ... I am apologetic for getting stuck on such a petite part of the fix 😓 ... Thank you @bari12 for all the input 😄

**Review by bari12 — APPROVED** [review]

**Comment by bari12** [comment]
Merging this as a feature for 1.26 due to error msg.

**Pull request merged**